subsys/pm/device.c

```diff
@@ -84,6 +126,7 @@ int pm_device_action_run(const struct device *dev,
 	int ret;
 	enum pm_device_state state;
 	struct pm_device *pm = dev->pm;
+	bool domain;
```

Comment:
note: this function is used by system PM. Consider this scenario: PD -> DEV during suspension.

- DEV is suspended
- PD is suspended. DEV is a dependent, so it will be notified with `PM_DEVICE_ACTION_HANDLE_PD_SUSPEND`. However, the device is suspended.

So my question: if I'm a device, under which conditions I'll receive a `PM_DEVICE_ACTION_HANDLE_PD_SUSPEND`? If I can get that notification when I'm both suspended and active, the actions I can/would take are likely different, so this needs some clarification IMO.

Reply:
It is up to device decide what to do in this case, they have their own particular behavior, we cannot stipulate what they should do. What can be done is to make explicit that this workflow happens and that they should be aware.